### PR TITLE
Make models not embeddable

### LINKED ISSUE #20815: Models should not be embeddable
**Describe the bug**
Talking with the product and @brunobergher specifically, it turns out that models should not be embeddable objects. They currently are but that should change.

**Logs**
N/A

**To Reproduce**
1. Turn any saved question into a model
2. Notice the share icon in the lower right corner
    - If you click on it, it's possible to embed and share that model
 
**Expected behavior**
Share icon shouldn't exist for models in the first place.

**Screenshots**
N/A

**Information about your Metabase Installation:**
- local dev, `master`, 0b06f460344afdefd3b127140cc96aa58a404e25, H2, Sample database
- also release x.42.2

**Severity**
?

### PR DESCRIPTION
**Pull request by WiNloSt**
Fixes #20815

### CODE REVIEW

File: frontend/src/metabase/query_builder/containers/QuestionEmbedWidget.jsx [hunk 1]
```diff
@@ -92,6 +92,10 @@ export default class QuestionEmbedWidget extends Component {
     isPublicLinksEnabled = MetabaseSettings.get("enable-public-sharing"),
     isEmbeddingEnabled = MetabaseSettings.get("enable-embedding"),
   }) {
+    if (question.isDataset()) {
+      return false;
+    }
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/containers/QuestionEmbedWidget.jsx:97]: I'm not sure if we should only disable *Embed in other applications* but not public embed. But I'm following the repro, which only expects the button to not be visible.

Comment (nemanjaglumac) [reply frontend/src/metabase/query_builder/containers/QuestionEmbedWidget.jsx:97]: I am sorry for the late reply @WiNloSt. Tbh, I am also not sure to which extent do we want to deal with models and embedding. It seems like that is still being talked about, so I'm hoping @brunobergher could be of more help than I am.

Comment (flamber) [reply frontend/src/metabase/query_builder/containers/QuestionEmbedWidget.jsx:97]: Embedding = Signed Embedding + Public Sharing = so both, hide the icon.

Comment (WiNloSt) [reply frontend/src/metabase/query_builder/containers/QuestionEmbedWidget.jsx:97]: > Embedding = Signed Embedding + Public Sharing = so both, hide the icon.

Thanks!

**Review by kulyk — APPROVED** [review]
Looks good! I'd wait for an approval from @flamber or @nemanjaglumac though, just to be completely sure

**Review by flamber — APPROVED** [review]

**Pull request merged**